vegbilder-client/src/components/ImageView/History/History.tsx

```diff
@@ -117,19 +128,74 @@ const getDateAndTimeString = (imagePoint: IImagePoint) => {
   return `${dateTime?.date} kl. ${dateTime?.time}`;
 };
 
-const imagePointsAreOnSameVegkategori = (imagePointA: IImagePoint, imagePointB: IImagePoint) => {
-  return imagePointA.properties.VEGKATEGORI === imagePointB.properties.VEGKATEGORI;
-};
-
 interface IHistoryProps {
   setIsHistoryMode: (isHistoryMode: boolean) => void;
 }
 
+async function findHistoryImagePoints(
+    currentImagePoint: IImagePoint | null, 
+   loadedImagePoints: ILoadedImagePoints | null, 
+   historyImagePoints: IImagePoint[], setHistoryImagePoints: React.Dispatch<React.SetStateAction<IImagePoint[]>>, 
+   availableYearsForAllImageTypes: availableYears) 
+   {
+    if (currentImagePoint && loadedImagePoints) {
+    const shouldNotRecalcualteHistoryImages = historyImagePoints.includes(currentImagePoint);
+      if (!shouldNotRecalcualteHistoryImages) {
+      setHistoryImagePoints([]);
+      const currentCoordinates = getImagePointLatLng(currentImagePoint);
+      const currentImagePointTime = getDateObj(currentImagePoint).getTime();
+
+      setHistoryImagePoints((prevState) => [currentImagePoint, ...prevState]);
+
+      if (currentCoordinates) {
+        const bbox = {
+          west: currentCoordinates.lng,
+          south: currentCoordinates.lat,
+          east: currentCoordinates.lng, // create the smallest possible bbox area
+          north: currentCoordinates.lat,
+        };
+
+        for (const [imageType, years] of Object.entries(availableYearsForAllImageTypes)) {
+          if (imageType === 'all') {
+            return;
+          }
+          for (const year of years) {
+            let typename = imageType === 'panorama' ? `vegbilder_1_0:Vegbilder_360_${year}` : `vegbilder_1_0:Vegbilder_${year}`;
+            await getImagePointsInTilesOverlappingBbox(bbox, typename).then((res) => {
+              const imagePoints = res.imagePoints;
+              const uniqueDatesInOtherYears: Set<number> = new Set();
+              const imagePointsGroupedByTime: Dictionary<IImagePoint[]> = groupBy(
+                imagePoints,
+                (imagePoint: IImagePoint) => {
+                  const time = getDateObj(imagePoint).getTime();
+                  if (time !== currentImagePointTime) {
+                    uniqueDatesInOtherYears.add(time);
+                  }
+                  return time;
+                }
+              );
```

Comment:
```suggestion
                const imagePointsGroupedByTime: NumericDictionary<IImagePoint[]> = groupBy(
                imagePoints,
                (imagePoint: IImagePoint) => getDateObj(imagePoint).getTime()
              );
              const uniqueDatesInOtherYears: Set<number> = new Set(
                Object.keys(imagePointsGroupedByTime)
                  .map((x) => Number(x))
                  .filter((x) => x !== currentImagePointTime)
              );
```
Eg synest forslaget over er penare kode, men det er litt kronglete å forklare kvifor med tekst, så vi kan gjerne ta ein prat muntleg om endringa over.